ros_ws/src/autonomous/src/wall_following.cpp

```diff
@@ -1,202 +1,90 @@
 #include "wall_following.h"
+#include <boost/algorithm/clamp.hpp>
 
 WallFollowing::WallFollowing()
 {
-    lidar_subscriber =
+    this->m_lidar_subscriber =
         m_node_handle.subscribe<sensor_msgs::LaserScan>(TOPIC_LASER_SCAN, 1, &WallFollowing::lidarCallback, this);
-    emer_stop_subscriber =
-        m_node_handle.subscribe<std_msgs::Bool>(TOPIC_EMER_STOP, 1, &WallFollowing::emergencyStopCallback, this);
-    pid_publisher = m_node_handle.advertise<drive_msgs::pid_input>(TOPIC_PID_INPUT, 1);
+    this->m_emergency_stop_subscriber =
+        m_node_handle.subscribe<std_msgs::Bool>(TOPIC_EMERGENCY_STOP, 1, &WallFollowing::emergencyStopCallback, this);
+    this->m_drive_parameter_publisher = m_node_handle.advertise<drive_msgs::drive_param>(TOPIC_DRIVE_PARAMETERS, 1);
 }
 
-float WallFollowing::rangeAtDegree(const sensor_msgs::LaserScan::ConstPtr& lidar, float theta)
+float map(float in_lower, float in_upper, float out_lower, float out_upper, float value)
 {
-    // check if calculated index is invalid
-    // if so then return the chosen max range (here 4)
-    if (theta / 0.375f < 0 || theta / 0.375f > 719)
-    {
-        return 4;
-    }
-
-    // ranges size is 720, angle range is 270°
-    // so one step equals an angle of 0,375°
-    // to access range at angle theta, you have to divide the given theta by the step size
-    float range = lidar->ranges[theta / 0.375];
-
-    // check for absurd or nan values
-    // the max range should be 4 so the car doesn't get affected by walls very far away
-    if (range > 4 || std::isinf(range))
-    {
-        range = 4;
-    }
-    return range;
+    return out_lower + (out_upper - out_lower) * (value - in_lower) / (in_upper - in_lower);
```

Comment:
Not necessary but makes it clearer on first glance:
```suggestion
    return out_lower + ((out_upper - out_lower) * (value - in_lower) / (in_upper - in_lower));
```